### ISSUE TITLE
Use SparkSession in Spark operations

### ISSUE THREAD

**Issue by gaffer01** [gchq/Gaffer#1121]
The operation `GetRDDOfElements` requires a `SparkContext`; the operation `GetJavaRDDOfElements` requires a `JavaSparkContext`; the operation `GetDataFrameOfElements` requires an `SQLContext`. It would be more consistent to require a `SparkSession` for all of these and obtain a `SparkContext` or `JavaSparkContext` from that as necessary.

This should be done before version 1 as it will cause users to have to change the code they use to get Spark RDDs/Dataframes.

**Comment by m55624** [comment]
# Migration Notes

All Spark RDD Operations now require a SparkSession instead of specific contexts.  When creating any of these Operations the constructor will now use a SparkSession.

This means when doing any Operations through the AbstractGetRDDHandler, the Operation passed in to the handler will now contain the SparkSession.

**Comment by p013570** [comment]
Merged into develop-1.0.0.
